accepted/0000-automatic-migration-salt-bundle.md

```diff
@@ -0,0 +1,62 @@
+- Feature Name: Automatic migration from Salt 3000 to Salt Bundle
+- Start Date: 2023-03-22
+
+# Summary
+[summary]: #summary
+
+Force an automatic migration for Salt clients from Salt 3000 to Salt Bundle.
+
+
+# Motivation
+[motivation]: #motivation
+
+
+Salt 3000 is EOL as of Aug 31, 2022. It is advisable for Salt clients using Salt 3000 to migrate to Salt Bundle. Right now this migration can be done by applying the already existent `util.mgr_switch_to_venv_minion` state from the CLI. The puprose of this RFC is to automatically perform this migration for Salt 3000 clients to Salt Bundle so they can keep receiving updates for the Salt package, including security updates.
+
+
+# Detailed design
+[design]: #detailed-design
+
+An state to perform the migration from Salt 3000 to Salt Bundle is added to the highstate. This state checks that the 'saltversion' grain in the targeted minion and that the minion is not already running Salt Bundle. In that case, the migration is performed in the following steps:
+
+- Install the `venv-salt-minion` package.
+- Copy the minion id file, minion configuration files, PKI files and static grains for the current `salt-minion` in `/etc/salt` to the `venv-salt-minion` corresponding directories in `/etc/venv-salt-minion`.
+- Enable the `venv-salt-minion` service.
+- Disable the `salt-minion` service.
+
+The whole process is transparent to the user.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- The `salt-minion` package remains installed and the configuration files in the `/etc/salt` are not deleted.
+
+- After migration this check is done in every highstate.
+
+- If a client was bootstrapped avoiding the Salt Bundle following the steps in the documentation, the migration would force this client to use Salt Bundle. Those steps are:
```

Comment:
Long term, we should only support bundle, and all this options should go away.

Reply:
Should this avoid-venv-salt-minion  feature be removed in the context of this RFC or it is something do in the future?

Reply:
Pablo statement looks good to me.